draft-ietf-mls-protocol.md

```diff
@@ -2321,23 +2369,26 @@ handle extensible fields:
   in the corresponding KeyPackage), then the client MUST reject the Welcome
   message and not join the group.
 
-* The extensions populated into a GroupContext object are drawn from those in
-  the GroupInfo object, according to the definitions of those extensions.
+* If any extension in a Commit message is unrecognized, then the client MUST
+  reject the Commit.  The sender of a Commit message MUST verify that the
+  extension is supported by all members of the group, as indicated in the
+  KeyPackages in the leaves of the tree.
+
+* The extensions populated into a GroupContext object by a joining member are
+  constructed from those in the GroupInfo object, according to the definitions
+  of those extensions.  Extensions in a Commit object may define modifications
+  to the list of GroupContext extensions, which recipients of the Commit apply
+  on receiving it.
```

Comment:
As I noted on the mailinglist, the issue with the PSKId being (part of) an extension and those extensions in turn being part of the `GroupInfo` object, means that the PSK can't be required for decrypting that `GroupInfo` object in the first place. In #336 , we suggested that the PSKId be part of the `GroupSecrets` object along with the other secrets. This would also ensure compatibility with the pending nPRF key schedule proposal.